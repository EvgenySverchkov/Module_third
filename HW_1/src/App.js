import React from "react";

import UserForm from "./components/userForm";
import Greetings from "./components/greetings";

export default class App extends React.Component{
	constructor(props){
		super(props);
		this.onSubmit = this.onSubmit.bind(this);
		this.onBack = this.onBack.bind(this);
		this.state = {isGrating: false};
	}

	onSubmit(data){
		this.setState({isGrating: true});
		this.data = data;
	}

	onBack(){
		this.setState({isGrating: false});
	}
	
	render(){
		const isGrating = this.state.isGrating;
		let component = null;

		if(isGrating){
			component = <Greetings data={this.data} onBack={this.onBack}/>;
		}
		else{
			component = <UserForm formHeader={this.onSubmit}/>;
		}
		return <div>{component}</div>;
	}
}

