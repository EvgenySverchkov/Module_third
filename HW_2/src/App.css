#root {
  width: 100%;
  min-height: 100%;
  justify-content: center;
  align-items: center;
  display: flex;
  flex-direction: column;
}