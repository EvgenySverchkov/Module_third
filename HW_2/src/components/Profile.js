import React from 'react';
import "./Profile.css";

export default class Profile extends React.Component{
	render(){
		return (<>
			<form className="form-inline" onSubmit={(e)=>{
				e.preventDefault();
				this.props.header(e.target.elements.userId.value);
			}}>
			<div className="form-group mb-2">
				<label forhtml='userId' className="col-lm-2 col-form-label">Input userId</label>
			</div>
			<div className="form-group mx-sm-3 mb-2">
				<input type="number" id="userId" className="form-control" placeholder="Eneter id"></input>
			</div>
			
			<button type="submit" className="btn btn-primary mb-2">Submit</button>
			</form>
		</>);
	}
}