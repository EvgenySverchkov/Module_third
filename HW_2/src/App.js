import React from 'react';
import Profile from "./components/Profile.js";
import UserInfoTable from "./components/UserInfoTable.js";
import "./App.css";

class App extends React.Component{
	constructor(props){
		super(props);
		this.onSubmit = this.onSubmit.bind(this);
		this.backHandler = this.backHandler.bind(this);
		this.state = {isTable:false};
	}
	onSubmit(userId){
		this.userId = userId;
		if(!this.userId){
			alert("You should enter userId to continue");
			this.setState({isTable:false});
		}else{
			this.setState({isTable:true});
		}
	}
	backHandler(){
		this.setState({isTable:false});
	}
	render(){
		const isTable = this.state.isTable;
		if(isTable){
			return <UserInfoTable idUser={this.userId} backHandler ={this.backHandler}/>;
		}
		
		else{
			return <Profile headerForSubmit={this.onSubmit}/>;
		}
	}
}

export default App;
