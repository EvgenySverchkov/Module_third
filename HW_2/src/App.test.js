import React from 'react';
import { shallow, mount } from 'enzyme';

import Profile from './components/Profile.js';
import App from './App.js';



describe("App tests", ()=>{
	it("snapshot test for App", ()=>{
		const wraper = shallow(<App />);
		expect(wraper.debug()).toMatchSnapshot();
	});

	it("test default component (Profile)", ()=>{
		const wraper = shallow(<App />);

		const profile = wraper.find('Profile');
		profile.simulate('submit');
		expect(wraper.state('isTable')).toEqual(false);
	});
});
