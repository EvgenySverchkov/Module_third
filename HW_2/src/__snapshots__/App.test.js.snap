// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests snapshot test for App 1`] = `"<Profile header={[Function: bound onSubmit]} />"`;
